File: numbers/prob1281.py
# Subtract the Product and Sum of Digits of an Integer

class Solution(object):
    def subtractProductAndSum(self, n):
        """
        :type n: int
        :rtype: int
        """
        digits = map(int, list(str(n)))

        from functools import reduce
        prod = reduce(lambda x, y : x * y, digits)
        add  = reduce(lambda x, y : x + y, digits)
        return prod - add

File: math/prob1281.kt
class Solution {
    fun subtractProductAndSum(n: Int): Int {
        
        var digit: Int
        var m = n
        var prod = 1 
        var tot = 0
        
        while (m > 0) {
            digit = m % 10
            prod *= digit
            tot += digit
            m = m / 10
        }

        return prod - tot
        
    }
}
